haystack/preview/components/file_converters/txt.py

```diff
@@ -0,0 +1,234 @@
+        metas = TextfileToDocument._prepare_metadata(data.meta, file_paths)
+
+        documents = []
+        for path, meta in tqdm(
+            zip(file_paths, metas), total=len(file_paths), desc="Converting text files", disable=not data.progress_bar
+        ):
+            text = TextfileToDocument._read_and_clean_file(
+                path=path, encoding=data.encoding, remove_numeric_tables=data.remove_numeric_tables
+            )
+
+            if data.valid_languages is not None:
+                if not TextfileToDocument._validate_language(text, data.valid_languages):
+                    logger.warning(
+                        "Text from file %s is not in one of the valid languages: %s. "
+                        "The file may have been decoded incorrectly.",
+                        path,
+                        data.valid_languages,
+                    )
+
+            document = Document(content=text, meta=meta, id_hash_keys=data.id_hash_keys)
+            documents.append(document)
+
+        return self.output(documents=documents)
+
+    @staticmethod
+    def _prepare_metadata(meta: Optional[Union[Dict, List[Dict]]], file_paths: List[Union[str, Path]]) -> List[Dict]:
+        """
+        Prepare the metadata for the Documents.
+
+        :param meta: The metadata for the Documents.
+        :param file_paths: The paths to the text files.
+        """
+        if meta is None:
+            return [{"file_path": str(path)} for path in file_paths]
+
+        if isinstance(meta, dict):
+            meta = [meta] * len(file_paths)
+
+        if len(meta) != len(file_paths):
+            raise PipelineRuntimeError(
+                f"The number of meta entries must match the number of paths if meta is a list. "
+                f"Number of paths: {len(file_paths)}, number of meta entries: {len(meta)}"
+            )
+
+        return [{**m, "file_path": m.get("file_path", str(path))} for m, path in zip(meta, file_paths)]
+
+    @staticmethod
+    def _read_and_clean_file(path: Union[str, Path], encoding: str, remove_numeric_tables: bool) -> str:
+        """
+        Read and clean the text file.
+
+        :param path: The path to the text file.
+        :param encoding: The encoding of the text file.
+        :param remove_numeric_tables: Whether to remove numeric tables.
+        """
+        if not Path(path).exists():
+            raise PipelineRuntimeError(f"File at path {path} does not exist.")
+
+        with open(path, encoding=encoding) as file:
+            text = file.read()
+            pages = text.split("\f")
+            cleaned_pages = [TextfileToDocument._clean_page(page, remove_numeric_tables) for page in pages]
+            return "\f".join(cleaned_pages)
+
+    @staticmethod
+    def _clean_page(page: str, remove_numeric_tables: bool) -> str:
+        """
+        Clean a page of text from numeric tables if `remove_numeric_tables` is `True`.
+
+        :param page: The page of text.
```

Comment:
Is this a page number? Or just text?

Reply:
This is the textual content of a page, I updated the docstring.